File: app/models/exam.rb
class Exam < ActiveRecord::Base
  belongs_to :user
  has_many :questions, -> { order("position ASC") }
  validates :name, presence: true
  validates :user, presence: true

  def total_points
    self.questions.inject(0) { |sum, q| sum + q.points }
  end
end

File: frontend/app/models/exam.js
import Ember from 'ember';
import DS from 'ember-data';

export default DS.Model.extend({
  name: DS.attr('string'),
  user: DS.belongsTo('user'),

  questions: DS.hasMany('question', { async: true }),

  totalPoints: Ember.computed('questions.@each.points', function() {
      var questions = this.get('questions');
      return questions.reduce(function(prevValue, question) {
          return prevValue + parseInt(question.get('points'));
      }, 0);
  }),
});
